### ISSUE TITLE
Little doubt

### ISSUE THREAD

**Issue by lelemm** [Genymobile/gnirehtet#280]
Your solution is awesome, I'm loving it. 
Is it possible to connect from the phone to a server @ the PC running gnirehtet?

I'm using it to share connection as supposed, but I do develop apps and backend servers in the pc running gnirehtet, so is it possible to access an API (HTTP) from android thru gnirehtet (the API is hosted at the pc running gnirehtet)?

Windows 10 here.

Thanks!

**Comment by rom1v** [comment]
> Is it possible to connect from the phone to a server @ the PC running gnirehtet?

Yes. From the device, `localhost` is the device itself, but `10.0.2.2` is the `localhost` of the computer.

https://github.com/Genymobile/gnirehtet/blob/789996167c4cdbcebaab9d0240837a5dabffe5a4/relay-rust/src/main.rs#L204

**Comment by lelemm** [comment]
Nice, thanks. Didn't see that in the description. My bad.
Ty!!

**Comment by lelemm** [comment]
Tried it out. Not sure if i'm doing something wrong, but when i try to connect to the API at 10.0.2.2, it gives this at catlog:

2020-03-11 11:16:43.690 30631-30651/? E/PersistentRelayTunnel: Cannot receive from tunnel
    java.io.IOException: Invalid argument

**Comment by lelemm** [comment]
Just a feedback.

I'm not sure if I will spend time investigating this, but it looks like that the IDENTITYSERVER4 (net core) does not like tunneling.

The authentication process works fine, but every api call that requires a bearer token, returns 401 ( I tried to use the token generated from the app at postman, and it returns 401 too)

![image](https://user-images.githubusercontent.com/15043428/76430614-465a0800-638f-11ea-88ca-b1db22959d7f.png)

Probably it would need some proxying to work properly.

**Comment by rom1v** [comment]
This token is sent by your app (via okhttp). It should be independant of gnirehtet (which just relays TCP packets).

**Comment by lelemm** [comment]
Found out the problem.
My authentication server is not published yet as an external service, so, the requested host is passed from my API to IdentityServer to check if the token is valid.
But since the IP 10.0.2.2 does not exists for windows, it understands as invalid.

Since I have a npcap loopback adapter, i set the IP 10.0.2.2 so the authentication service recognizes it as an valid ip:
![image](https://user-images.githubusercontent.com/15043428/76442477-c5efd300-639f-11ea-9d97-e5c07e415f0f.png)


After that, I changed the local configuration to the authority as below:

![image](https://user-images.githubusercontent.com/15043428/76442348-95a83480-639f-11ea-88ea-ab4a22a9802e.png)

Of course all of this is temporary so I can develop the app/api without using the local intranet or wifi routering.
